packages/factory/safe-singleton/src/index.ts

```diff
@@ -0,0 +1,58 @@
+import {
+    BaseSingletonFactory,
+    ProviderAdapter,
+    Transaction,
+} from "@gnosis.pm/singleton-deployer-core"
+import { getSingletonFactoryInfo, SingletonFactoryInfo } from "@gnosis.pm/safe-singleton-factory"
+
+// A factory based on https://github.com/gnosis/safe-singleton-factory
+// It has to be deployed to the destination network to use
+export class SafeSingletonFactory extends BaseSingletonFactory {
+    private factoryInfo: SingletonFactoryInfo
+
+    constructor(provider: ProviderAdapter, public chainId: number) {
+        super(provider)
+
+        const factoryInfo = getSingletonFactoryInfo(this.chainId)
+        if (!factoryInfo) {
+            throw new Error(
+                `No factory info found for chainId ${this.chainId}, please check instructions in https://github.com/gnosis/safe-singleton-factory on how to add it`
+            )
+        }
+        this.factoryInfo = factoryInfo
+    }
+
+    get address() {
+        return this.factoryInfo.address
+    }
+
+    get deploymentTx() {
+        return this.factoryInfo.transaction
+    }
+
+    get deploymentCosts() {
+        return "0x" + (this.factoryInfo.gasLimit * this.factoryInfo.gasPrice).toString(16)
+    }
+
+    get deployer() {
+        return this.factoryInfo.signerAddress
+    }
+
+    calculateSingletonAddress(
+        bytecode: string,
+        salt = "0x54Fe000000000000000000000000000000000000000000000000000000000000"
```

Comment:
why not 0? :D

Reply:
i am 54Fe pilled

Reply:
Only argument from my side for the 0 would be that this is the default when using hardhat. But as it would probably be hard to get the same deployment code for hardhat and truffle anyways it should not matter too much.